defaults/main.yml

```diff
@@ -116,6 +116,16 @@ checkmk_agent__user_home: '/var/lib/nagios'
 # SSH Public key for logging in with SSH user.
 checkmk_agent__user_key: ''
 
+
+# .. envvar:: checkmk_agent__authorized_keys__dependent_list
+#
+# Authorized key configuration for the ``debops.authorized_keys`` role.
+checkmk_agent__authorized_keys__dependent_list:
+  - name: '{{ checkmk_agent__ssh_user }}'
+    ssh_keys: [ '{{ checkmk_agent__user_key }} if {{ checkmk_agent__user_key|d() }} else {{ hostvars[groups["debops_service_checkmk_server"][0]].ansible_local.checkmk_server.ssh_publickey|d("") }}' ]
```

Comment:
The point is, that they key is generated during the setup of the monitoring site. If the user has to specify this manually, it would abort the playbook run, then the user has to lookup they key manually. Or he need to generate a keypair in advance and configure them before running the server setup.

I wish to find a solution where the setup just works. See also debops-contrib/ansible-checkmk_server#15